Docs: add GCP bring-your-own-VPC docs. (#2776)
skypilot-org/skypilot: docs/source/cloud-setup/cloud-permissions/gcp.rst

You can inspect and manage firewall rules at
``https://console.cloud.google.com/net-security/firewall-manager/firewall-policies/list?project=<your-project-id>``
or using any of GCP's SDKs.

.. _gcp-bring-your-vpc:

Using a specific VPC
-----------------------
By default, SkyPilot uses the following behavior to get a VPC to use for all GCP instances:

> - First, the VPC named ``default`` is checked against minimal recommended firewall rules for
>   SkyPilot to function. If it satisfies these rules, this VPC is used.

Reviewer: ```suggestion
- First, all existing VPCs in the project are checked against minimal recommended firewall rules for
  SkyPilot to function. If it satisfies these rules, this VPC is used.
```

Author: Good catch! Updated here + config.yaml docs too.
